### PR TITLE
Make the partner fee max cap configurable in autopilot

### LINKED ISSUE #2554: feat: Add configuration for partner fee cap in autopilot configuration
### Problem
We should make configurable in the autopilot the maximum partner fee amount.

### Acceptance criteria
The partner fee is capped to the amount specified in the autopilot configuration.

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
We should make configurable in the autopilot the maximum partner fee amount.

# Changes
- Introduce a new configuration parameter in the autopilot for the partner fee max cap with a default value 0.01
- Enforce the configured partner fee max cap

## How to test
1. e2e test
2. unit test

## Related Issues
Fixes #2554

### CODE REVIEW

**Comment by squadgazzz** [comment]
I'd suggest rebasing onto [this branch](https://github.com/cowprotocol/services/pull/2550) to avoid a lot of conflicts.

File: crates/autopilot/src/domain/fee/mod.rs
Comment (squadgazzz) [inline crates/autopilot/src/domain/fee/mod.rs:33]: Is it necessary to move this function here and also make fields crate-public?

File: crates/e2e/tests/e2e/protocol_fee.rs
Comment (squadgazzz) [inline crates/e2e/tests/e2e/protocol_fee.rs:374]: nit: don't really feel confident to use strings here. Would rather see a protocol fees config struct.

File: crates/autopilot/src/domain/fee/mod.rs
Comment (Mateo-mro) [reply crates/autopilot/src/domain/fee/mod.rs:33]: sure!

File: crates/e2e/tests/e2e/protocol_fee.rs
Comment (Mateo-mro) [reply crates/e2e/tests/e2e/protocol_fee.rs:374]: Sure! I am changing to accept anything which implements `ToString`. The reason of it is that we can keep the current `FeePolicyKind` enum without needing to convert it to string, and at the same time it adds customization to what you want to pass. 

The idea of doing it like that is to avoid creating a struct containing `FeePolicyKind` and a new field which is going to be used just is one test, and at the same time it opens to possibility of adding more tests with different configurations without altering the `execute_test()` signature.

**Comment by Mateo-mro** [comment]
> I'd suggest rebasing onto [this branch](https://github.com/cowprotocol/services/pull/2550) to avoid a lot of conflicts.

Thanks for the headsup! I saw the conflicts and aren't many. I will wait until your PR is merged so I don't pass the conflicts to you.

File: crates/autopilot/src/domain/fee/mod.rs
Comment (MartinquaXD) [inline crates/autopilot/src/domain/fee/mod.rs:54]: Just to double check. The fee policy extracted from the app data here will get stored in the DB when the order gets executed during an auction, right?

**Review by MartinquaXD — APPROVED** [review]
Code looks good to me.

File: crates/autopilot/src/domain/fee/mod.rs
Comment (fleupold) [inline crates/autopilot/src/domain/fee/mod.rs:58]: nit: I think it would be nice if we already panicked on startup in case `fee_policy_max_partner_fee` is >= 1. 
Otherwise, in case of bad configuration, we may have to wait until a user places an order with a partner fee (with some bad luck during the night on a weekend) before we run into a crash loop. It would be better to crashloop during the rollout.

**Review by fleupold — APPROVED** [review]
LG one comment

File: crates/autopilot/src/domain/fee/mod.rs
Comment (MartinquaXD) [reply crates/autopilot/src/domain/fee/mod.rs:58]: This should already be the case since the CLI arg gets parsed into a FeeFactor that enforces that invariant in its FromStr implementation.

Comment (Mateo-mro) [reply crates/autopilot/src/domain/fee/mod.rs:58]: @fleupold, as @MartinquaXD said, it is already imposed by the `FeeFactor` and it panics on startup in case of bad configuration.

**Review by sunce86 — APPROVED** [review]

**Review by squadgazzz — APPROVED** [review]
LGTM

**Review by sunce86 — APPROVED** [review]

**Pull request merged**
